solana/programs/swap-layer/src/processor/stage_outbound.rs

```diff
@@ -71,6 +71,12 @@ pub struct StageOutbound<'info> {
             // Cannot send to zero address.
             require!(args.recipient != [0; 32], SwapLayerError::InvalidRecipient);
 
+            // Min amount out must be specified for swaps into USDC.
+            require!(
+                args.min_amount_out.is_some() || src_mint.key() == common::USDC_MINT,
```

Comment:
So`|| src_mint.key() == common::USDC_MINT` will short-circuit this check. Is this relying on other constraints where the src mint and dst mint can't both be equal to 0?